tests/resources/Harbor-Util.robot

```diff
@@ -15,50 +15,53 @@
-
-    Log To Console  Downloading Harbor OVA...
-    ${out}=  Run  wget https://github.com/vmware/harbor/releases/download/${HARBOR_SHORT_VERSION}/${HARBOR_VERSION}.ova
-    Log To Console  Generating OVF file...
-    ${out}=  Run  ovftool ${HARBOR_VERSION}.ova ${HARBOR_VERSION}.ovf
-    Log To Console  Installing Harbor into test server...
-    ${hostString}=  Set Variable If  '${datacenter}' is not '${EMPTY}'  vi://${user}:${password}@${host}/${datacenter}/host/${cluster}
-    ...   '${datacenter}' is '${EMPTY}'  vi://${user}:${password}@${host}
-    ${out}=  Run  ovftool --noSSLVerify --acceptAllEulas --datastore=${datastore} --name=${name} --diskMode=thin --powerOn --X:waitForIp --X:injectOvfEnv --X:enableHiddenProperties --net:"Network 1"="${network}" --prop:vm.vmname=Harbor --prop:root_pwd='${password}' --prop:harbor_admin_password='${password}' --prop:db_password='${password}' --prop:auth_mode=db_auth --prop:permit_root_login=true --prop:verify_remote_cert=${verify} --prop:protocol=${protocol} ${HARBOR_VERSION}.ova '${hostString}'
+    Set Test Variable  ${host}  @{URLs}[${IDX}]
+
+    Log To Console  \nDeploying ova...
+    ${out}=  Run  ovftool --noSSLVerify --acceptAllEulas --datastore=${datastore} --name=${name} --net:"Network 1"='${network}' --diskMode=thin --powerOn --X:waitForIp --X:injectOvfEnv --X:enableHiddenProperties --prop:root_pwd=${password} --prop:harbor_admin_password=${password} --prop:db_password=${db_password} --prop:auth_mode=db_auth --prop:verify_remote_cert=${verify} --prop:protocol=${protocol} ${HARBOR_VERSION}.ova 'vi://${user}:${password}@${host}'
     Should Contain  ${out}  Received IP address:
+    Should Not Contain  ${out}  None
 
     ${out}=  Split To Lines  ${out}
     :FOR  ${line}  IN  @{out}
     \   ${status}=  Run Keyword And Return Status  Should Contain  ${line}  Received IP address:
     \   ${ip}=  Run Keyword If  ${status}  Fetch From Right  ${line}  ${SPACE}
-    \   Run Keyword If  ${status}  Set Environment Variable  HARBOR_IP  ${ip}
+    \   Run Keyword If  ${status}  Set Test Variable  ${harbor-ip}  ${ip}
     \   Exit For Loop If  ${status}
 
     Log To Console  Waiting for Harbor to Come Up...
-    :FOR  ${i}  IN RANGE  10
-    \  ${out}=  Run  curl -k https://%{HARBOR_IP}
-    \  Log    ${out}
+    :FOR  ${i}  IN RANGE  40
+    \  ${out}=  Run  curl -k ${protocol}://${harbor-ip}
+    \  Log  ${out}
     \  ${status}=  Run Keyword And Return Status  Should Not Contain  ${out}  502 Bad Gateway
-    \  Return From Keyword If  ${status}
-    \  Sleep  30
-    Fail  Harbor failed to come up properly!
+    \  ${status}=  Run Keyword If  ${status}  Run Keyword And Return Status  Should Not Contain  ${out}  Connection refused
+    \  ${status}=  Run Keyword If  ${status}  Run Keyword And Return Status  Should Contain  ${out}  <title>Harbor</title>
+    \  Run Keyword If  ${status}  Set Test Variable  ${status}  ${status}
```

Comment:
they're overwriting ${status} based on a few situations I observed in practice since 'should not contain  502 bad gateway' alone doesn't guarantee harbor is up.